src/rgw/services/svc_sys_obj_core.cc

```diff
@@ -623,12 +616,9 @@ int RGWSI_SysObj_Core::pool_list_objects_init(const DoutPrefixProvider *dpp,
 
   auto& ctx = static_cast<PoolListImplInfo&>(*_ctx->impl);
 
-  ctx.pool = rados_svc->pool(pool);
-  ctx.op = ctx.pool.op();
-
-  int r = ctx.op.init(dpp, marker, &ctx.filter);
```

Comment:
this no longer initializes the `marker`, so the failed user listing was starting at the beginning instead of resuming from the previous position. candidate fix in https://github.com/ceph/ceph/pull/54748